client/blocks/post-edit-button/index.jsx

```diff
@@ -0,0 +1,33 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import Gridicon from 'components/gridicon';
+import { localize } from 'i18n-calypso';
+import { getEditURL } from 'lib/posts/utils';
+
+const PostEditButton = ( { post, site, size, translate } ) => {
+	const editUrl = getEditURL( post, site );
+	return (
+		<a className="post-edit-button" href={ editUrl }>
+			<Gridicon icon="pencil" size={ size } className="post-edit-button__icon" />
+			<span className="post-edit-button__label">{ translate( 'Edit' ) }</span>
+		</a>
+	);
+};
+
+PostEditButton.propTypes = {
+	post: React.PropTypes.object.isRequired,
```

Comment:
what would you think about calling these `postId` and `siteId` so as to make it clear that they're just IDs and not the full post / site objects?